staging/src/k8s.io/kubectl/pkg/cmd/run/run.go

```diff
@@ -306,6 +375,15 @@ func (o *RunOptions) Run(f cmdutil.Factory, cmd *cobra.Command, args []string) e
 		return err
 	}
 
+	//add remaining needed options generated during validation
+	o.restartPolicy = restartPolicy
+	o.timeout = timeout
+	o.remove = remove
+
+	return nil
+}
+
+func (o *RunOptions) Run(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
```

Comment:
This is an issue. The `Run` function should not reference `cobra`.

Please have a look at the `wait` command, in which the `WaitOptions.Run()` does not have any arguments.